repository-management.md

```diff
@@ -0,0 +1,101 @@
+* `framework` - Part of the WP-CLI framework itself, which provides the architecture, API and helper functions to make commands possible.
+* `testing` - Part of the unit or functional tests.
+* `website` - Part of the website infrastructure on `wp-cli.org` or `make.wordpress.org`.
+
+_Example: `scope:documentation`_
+
+#### State
+
+The labels that defined what state a given issue/pull request is in are prefixed with `state:`.
+
+Used states:
+
+* `unconfirmed` - The bug/problem in the issue could not be replicated yet or might be related to the reporter's environment.
+* `unsupported` - The issue is outside of the scope of a bug report and cannot be supported on GitHub. The reporter should be pointed towards one of the [support channels](http://wp-cli.org/#support). 
+
+_Example: `state:unconfirmed`_
+
+### Required Labels
+
+Some labels have a special meaning and/or might be used for automated workflows down the road. These are required across all official packages.
+
+#### Bug
+
+The `bug` label denotes an issue that is a confirmed bug where the code does not produce the expected result.
+
+#### Good First Issue
+
+The `good-first-issue` label denotes an issue that is a good entry point for new contributors wanting to get their feet wet.
+
+Good first issues are small in scope and don't require extensive technical expertise or historical project knowledge to get started.
+
+## Commits
+
+No direct commits are to be done against the `master` branch of the packages. All code changes need to go through a pull request workflow. 
+
+## Pull Requests
+
+All code changes go through a pull request.
+
+Every submitted pull request needs to go through a code review, and needs to be approved by at least one of the [committers](https://github.com/orgs/wp-cli/teams/committers).
+
+Non-trivial pull requests should be preceded by a related issue that defines the problem to solve and allows for discussion of the most appropriate solution before actually writing code.
+  
+When a pull request was submitted by one of the [committers](https://github.com/orgs/wp-cli/teams/committers), the submitter should set the "Reviewers" for that pull request to `wp-cli/committers` if a general code review is needed, or to one or more specific committer profiles if the expertise of a specific person is needed/wanted.
+
+When a pull request was submitted by an external contributor, the [committers](https://github.com/orgs/wp-cli/teams/committers) should be responsive and provide quick feedback to encourage further contributions.
+
+Apart from needing to be approved, pull requests also need to have their tests in a passing state before they can be merged. When both of these conditions are true, any [committer](https://github.com/orgs/wp-cli/teams/committers) can merge the pull requestby:
+1. Ensuring that all applicable labels have been set.
+2. Ensuring that the correct milestone has been set.
+3. Ensuring that the branch is deleted after the merge (if applicable).
```

Comment:
On the topic of pull requests, another thing I do at the end is re-title the pull request as needed:

![image](https://user-images.githubusercontent.com/36432/29568905-b2b0ec20-8706-11e7-9962-419d3726c452.png)

The ideal pull request title is one that I can more or less copy and paste into release notes.

Reply:
https://github.com/wp-cli/handbook/pull/127/commits/c286eb6eb0bcdf82bdf2a9cb36eb7cb9ba40218b